### DISCUSSION TITLE
Does invalidateQueries method accept a list of keys?

### DISCUSSION THREAD

**Discussion by vineetdigit** [TanStack/query#1950] [Q&A] [answered]
It appears that I can't pass multiple keys to `queryClient.invalidateQueries`. For example in following code, query for only key1 gets invalidated.

```
let queryKeys = [];
queryKeys.push(key1)
queryKeys.push(key2)

queryClient.invalidateQueries(...queryKeys, {})
```
Can someone confirm that this is the expected behavior? 

To work around, I'm calling `invalidateQueries` once per key. 

```
queryKeys.map(key => {
      queryClient.invalidateQueries(key, {})
});

```

**Comment by TkDodo** [comment, marked as answer]
> Does invalidateQueries method accept a list of keys?

it does not.

it excepts one key, which will be _fuzzily_ matched against the cache. see: https://react-query.tanstack.com/guides/query-invalidation#query-matching-with-invalidatequeries
